### PR TITLE
Fix put without response body

### PR DESCRIPTION
**Pull request by ArcturusZhang**
# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: samples/Azure.Management.Storage/Generated/BlobContainerContainer.cs [hunk 1]
```diff
@@ -45,26 +45,28 @@ internal BlobContainerContainer(ResourceOperationsBase parent) : base(parent)
...
         /// <param name="containerName"> The name of the blob container within the specified storage account. Blob container names must be between 3 and 63 characters in length and use numbers, lower-case letters and dash (-) only. Every dash (-) character must be immediately preceded and followed by a letter or number. </param>
         /// <param name="blobContainer"> Properties of the blob container to create. </param>
-        /// <param name="cancellationToken"> A token to allow the caller to cancel the call to the service. The default value is <see cref="CancellationToken.None" />. </param>
-        public virtual Response<BlobContainer> CreateOrUpdate(string containerName, BlobContainerData blobContainer, CancellationToken cancellationToken = default)
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="containerName"/> or <paramref name="blobContainer"/> is null. </exception>
+        public Response<BlobContainer> Create(string containerName, BlobContainerData blobContainer, CancellationToken cancellationToken = default)
```
Comment (ArcturusZhang) [inline samples/Azure.Management.Storage/Generated/BlobContainerContainer.cs:53]: make a test to see if overriding the resource will return errors

File: samples/Azure.ResourceManager.Sample/Generated/AvailabilitySetContainer.cs [hunk 2]
```diff
@@ -44,26 +44,28 @@ internal AvailabilitySetContainer(ResourceOperationsBase parent) : base(parent)
...
 
-        /// <summary> The operation to create or update a AvailabilitySet. Please note some properties can be set only during creation. </summary>
+        /// <summary> Create or update an availability set. </summary>
         /// <param name="availabilitySetName"> The name of the availability set. </param>
         /// <param name="parameters"> Parameters supplied to the Create Availability Set operation. </param>
-        /// <param name="cancellationToken"> A token to allow the caller to cancel the call to the service. The default value is <see cref="CancellationToken.None" />. </param>
-        public virtual Response<AvailabilitySet> CreateOrUpdate(string availabilitySetName, AvailabilitySetData parameters, CancellationToken cancellationToken = default)
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="availabilitySetName"/> or <paramref name="parameters"/> is null. </exception>
+        public Response<AvailabilitySet> CreateOrUpdate(string availabilitySetName, AvailabilitySetData parameters, CancellationToken cancellationToken = default)
```
Comment (m-nash) [inline samples/Azure.ResourceManager.Sample/Generated/AvailabilitySetContainer.cs:52]: these should be virtual

File: samples/Azure.ResourceManager.Sample/Generated/VirtualMachineExtensionImageContainer.cs [hunk 3]
```diff
@@ -50,7 +50,7 @@ internal VirtualMachineExtensionImageContainer(ResourceOperationsBase parent) :
         /// <param name="type"> The String to use. </param>
         /// <param name="version"> The String to use. </param>
         /// <param name="cancellationToken"> A token to allow the caller to cancel the call to the service. The default value is <see cref="CancellationToken.None" />. </param>
-        public virtual Response<VirtualMachineExtensionImage> Get(string location, string publisherName, string type, string version, CancellationToken cancellationToken = default)
+        public Response<VirtualMachineExtensionImage> Get(string location, string publisherName, string type, string version, CancellationToken cancellationToken = default)
```
Comment (m-nash) [inline samples/Azure.ResourceManager.Sample/Generated/VirtualMachineExtensionImageContainer.cs:53]: these should be virtual

File: test/TestProjects/ExactMatchFlattenInheritance/Generated/AzureResourceFlattenModel1Container.cs [hunk 4]
```diff
@@ -43,26 +43,27 @@ internal AzureResourceFlattenModel1Container(ResourceOperationsBase parent) : ba
...
         // Container level operations.
 
-        /// <summary> The operation to create or update a AzureResourceFlattenModel1. Please note some properties can be set only during creation. </summary>
         /// <param name="name"> The String to use. </param>
         /// <param name="parameters"> The AzureResourceFlattenModel1 to use. </param>
-        /// <param name="cancellationToken"> A token to allow the caller to cancel the call to the service. The default value is <see cref="CancellationToken.None" />. </param>
-        public virtual Response<AzureResourceFlattenModel1> CreateOrUpdate(string name, AzureResourceFlattenModel1Data parameters, CancellationToken cancellationToken = default)
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="name"/> or <paramref name="parameters"/> is null. </exception>
+        public Response<AzureResourceFlattenModel1> Put(string name, AzureResourceFlattenModel1Data parameters, CancellationToken cancellationToken = default)
```
Comment (m-nash) [inline test/TestProjects/ExactMatchFlattenInheritance/Generated/AzureResourceFlattenModel1Container.cs:50]: If a resource named their create method incorrectly we should rename it to CreateOrUpdate.  We shouldn't rename all PUT methods but there is one that represents CreateOrUpdate behavior and in this case it appears to be that way

**Review by m-nash — CHANGES_REQUESTED** [review]

File: test/TestProjects/ExactMatchFlattenInheritance/Generated/AzureResourceFlattenModel1Container.cs (hunk 4 above)
Comment (ArcturusZhang) [reply test/TestProjects/ExactMatchFlattenInheritance/Generated/AzureResourceFlattenModel1Container.cs]: I have changed this back

File: samples/Azure.Management.Storage/Generated/BlobContainerContainer.cs [hunk 5]
```diff
@@ -211,7 +216,7 @@ public async virtual Task<Response<BlobContainer>> GetAsync(string containerName
         /// <summary> Tries to get details for this resource from the service. </summary>
         /// <param name="containerName"> The name of the blob container within the specified storage account. Blob container names must be between 3 and 63 characters in length and use numbers, lower-case letters and dash (-) only. Every dash (-) character must be immediately preceded and followed by a letter or number. </param>
         /// <param name="cancellationToken"> A token to allow the caller to cancel the call to the service. The default value is <see cref="CancellationToken.None" />. </param>
-        public virtual BlobContainer TryGet(string containerName, CancellationToken cancellationToken = default)
+        public BlobContainer TryGet(string containerName, CancellationToken cancellationToken = default)
```
Comment (m-nash) [inline samples/Azure.Management.Storage/Generated/BlobContainerContainer.cs:219]: these need to be virtual

**Review by m-nash — CHANGES_REQUESTED** [review]

File: src/AutoRest.CSharp/Mgmt/Generation/MgmtClientBaseWriter.cs [hunk 6]
```diff
@@ -126,9 +136,9 @@ protected internal string AwaitKeyword(bool async)
             return async ? "await" : string.Empty;
         }
 
-        protected internal string OverrideKeyword(bool isInheritedMethod)
+        protected internal string OverrideKeyword(bool isInheritedMethod, bool isVirtual = false)
         {
-            return isInheritedMethod ? "override" : string.Empty;
+            return isInheritedMethod ? "override" : (isVirtual ? "virtual" : string.Empty);
```
Comment (m-nash) [inline src/AutoRest.CSharp/Mgmt/Generation/MgmtClientBaseWriter.cs:141]: shouldn't we call the method on line 124?

File: test/TestProjects/ExactMatchFlattenInheritance/Generated/AzureResourceFlattenModel2Container.cs [hunk 7]
```diff
@@ -43,26 +43,27 @@ internal AzureResourceFlattenModel2Container(OperationsBase parent) : base(paren
 
         // Container level operations.
 
-        /// <summary> The operation to create or update a AzureResourceFlattenModel2. Please note some properties can be set only during creation. </summary>
```
Comment (m-nash) [inline test/TestProjects/ExactMatchFlattenInheritance/Generated/AzureResourceFlattenModel2Container.cs:46]: why is the summary removed here?

**Review by m-nash — CHANGES_REQUESTED** [review]

File: src/AutoRest.CSharp/Mgmt/Generation/MgmtClientBaseWriter.cs (hunk 6 above)
Comment (ArcturusZhang) [reply src/AutoRest.CSharp/Mgmt/Generation/MgmtClientBaseWriter.cs:141]: Ah, I forgot we have that function. will change

File: test/TestProjects/ExactMatchFlattenInheritance/Generated/AzureResourceFlattenModel2Container.cs (hunk 7 above)
Comment (ArcturusZhang) [reply test/TestProjects/ExactMatchFlattenInheritance/Generated/AzureResourceFlattenModel2Container.cs:46]: The corresponding operation in the swagger does not have a description
I have added some to this swagger.
But I am not sure if we have more of this kind of cases in the test projects...

**Review by m-nash — APPROVED** [review]

**Pull request merged**